### ISSUE TITLE
Example torch_enet.cpp doesn't work

### ISSUE THREAD

**Issue by husaka** [opencv/opencv#10045]
##### System information (version)
- OpenCV => 3.3
- Operating System / Platform => Windows 10, 64 Bit
- Compiler => Visual Studio 2015

##### Detailed description

Example from "sources\samples\dnn\torch_enet.cpp doesn't work.
During execution occurs an runtime error.

##### Steps to reproduce
Command line args:
-model=".\ENet-CityScapes\model-best.net" -image=".\images\lion.jpg"

CommandLineParser reads args correctly (verified in debug)
Model reads normally: 
    dnn::Net net = dnn::readNetFromTorch(modelFile);
Net Outputs(1): l367_Deconvolution
Image reads normally.
Fails on
    Mat result = net.forward(oBlob);
"Unhandled exception at 0x00007FFCD02210CB (ucrtbased.dll) in TorchENetOpenCV.exe: An invalid parameter was passed to a function that considers invalid parameters fatal."

Thanks for the reply.

**Comment by alalek** [comment]
> 3.3

Try the latest release "OpenCV 3.3.1".

Please add (in priority list):
- stack trace
- local variables of last called OpenCV function
- input image dimensions / type

**Comment by husaka** [comment]
#### Version
OpenCV 3.3.1

#### Call Stack
ucrtbased.dll!00007ffcc78bf99d()	Unknown
 	ucrtbased.dll!00007ffcc78be855()	Unknown
 	msvcp140d.dll!00007ffcd8160801()	Unknown
 	opencv_world331d.dll!00007ffcba75c242()	Unknown
 	opencv_world331d.dll!00007ffcbb2b0063()	Unknown
 	opencv_world331d.dll!00007ffcba97635c()	Unknown
 	opencv_world331d.dll!00007ffcba9764e6()	Unknown
 	opencv_world331d.dll!00007ffcba977062()	Unknown
 	opencv_world331d.dll!00007ffcba975eff()	Unknown
 	opencv_world331d.dll!00007ffcba9760bf()	Unknown
 	opencv_world331d.dll!00007ffcba973752()	Unknown
 	concrt140d.dll!00007ffcd78c0a8b()	Unknown
 	concrt140d.dll!00007ffcd78da93a()	Unknown
 	concrt140d.dll!00007ffcd7908d1f()	Unknown
 	concrt140d.dll!00007ffcd78d60c6()	Unknown
 	concrt140d.dll!00007ffcd78d5f1d()	Unknown
 	concrt140d.dll!00007ffcd78d4102()	Unknown
 	concrt140d.dll!00007ffcd79124cb()	Unknown
 	kernel32.dll!00007ffd06382774()	Unknown
 	ntdll.dll!00007ffd08bb0d51()	Unknown

#### Registers
	RAX = 0000000000000000 RBX = 0000000000000000 RCX = 3F4CE813B9CF0000 RDX = 0000000000000000 
	RSI = 0000000000000000 RDI = 00000071244FE6A0 R8  = 0000020EA4FB3480 R9  = 0000000000000003 
	R10 = 0000020EA6E3E0C0 R11 = 00000071244F65A0 R12 = 0000000000000000 R13 = 0000000000000000 
	R14 = 0000000000000000 R15 = 0000000000000000 RIP = 00007FFCC78BF99D RSP = 00000071244F65B0 
	RBP = 0000000000000000 EFL = 00000244 

#### Local variables of last called OpenCV function
net	{impl={owner=0x0000023f7c26a760 <Information not available, no symbols loaded for opencv_world331d.dll> ...} }	cv::dnn::experimental_dnn_v2::Net

inputBlob	{flags=1124024325 dims=4 rows=-1 ...}	cv::Mat
		flags	1124024325	int
		dims	4	int
		rows	-1	int
		cols	-1	int
data	0x0000023f06009080 ...	unsigned char *
datastart	0x0000023f06009080 ...	const unsigned char *
dataend	0x0000023f06609080 ...  const unsigned char *
datalimit	0x0000023f06609080 ...	const unsigned char *
allocator	0x0000000000000000 <NULL>	cv::MatAllocator *
u	0x0000023f7e3d99e0 {prevAllocator=0x0000023f7c2736b0 <Information not available, no symbols loaded for opencv_world331d.dll> ...}	cv::UMatData *
size	{p=0x0000023f7e19c964 {1} }	cv::MatSize
step	{p=0x0000023f7e19c940 {6291456} buf=0x000000693b10eac0 {0, 0} }	cv::MatStep

#### Input image
flags	1124024336	int
dims	2	int
rows	512	int
cols	1024	int

type - 3 channels (CV_8UC3)

**Comment by husaka** [comment]
@alalek please tell why the issue is marked as incomplete, what I need to specify in addition?

**Comment by alalek** [comment]
Because there is no complete reproducer (with required input data) or information about clean reason of mentioned problem (stacktrace is not usable).

**Comment by jgoenetxea** [comment]
I also have an error with this sample, but it is not an execution error. In my case, the sample apparently 'works' but the result is wrong.

It looks like a matrix is not initialized during the execution.
This is the command I have used for my test (from the opencv compilation folder):

`
bin/example_dnn-torch_enet 
-c="/home/user/libs/opencv-3.3.1/data/enet/categories.txt" 
-m="/home/user/libs/opencv-3.3.1/data/enet/model-best.net" 
-r="/home/user/libs/Libraries/opencv-3.3.1/data/enet/result.blob" 
-i="/home/user/libs/Libraries/opencv-3.3.1/data/enet/image.jpg" 
--show
`

Check if you have included all the needed files. In my case, the image is downloaded from the internet:https://i.pinimg.com/736x/65/1d/e1/651de1af7ab0669a2bd4f1a016cfe24e.jpg 

However, the result is not correct at all. The output image looks like not segmented, or like every pixel belongs to the same category.

Analyzing this issue, I have realized that the 'categories.txt' file could not be right. This is because in this file assigns the value '1' to each category, and the function used to load the categories ('readColors') is expecting three values after the category name.

I have included some changes to that file to fit with the expectations of the 'readColors' function, but the result is still confusing.

It looks like the code of the sample is not up to date. Any suggestion in this matter is welcome.

Note: This is the content of the 'categories.txt' file after include color codes:

> Unlabeled 0 0 0
Road  128 0 0
Sidewalk 0 128 0
Building 128 128 0
Wall 0 0 128
Fence 128 0 128
Pole 0 128 128
TrafficLight 128 128 128
TrafficSign 64 0 0
Vegetation 192 0 0
Terrain 64 128 0
Sky 192 128 0
Person 64 0 128
Rider 192 0 128
Car 64 128 128
Truck 192 128 128
Bus 0 64 0
Train 128 64 0
Motorcycle 0 192 0
Bicycle 128 192 0
